sktime/annotation/adapters/_pyod.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+import numpy as np
+from sktime.annotation.base._base import BaseSeriesAnnotator
+
+__author__ = ["mloning", "satya-pattnaik", "fkiraly"]
+
+import pandas as pd
+
+
+class SeriesAnnotatorPyOD(BaseSeriesAnnotator):
+    """Transformer that applies outlier detector from pyOD
+
+    Parameters
+    ----------
+    estimator : PyOD estimator
+        See ``https://pyod.readthedocs.io/en/latest/`` documentation for a detailed
+        description of all options.
+    annotation_format : str - one of "sparse" and "dense"
+        "sparse" - sub-series of outliers is returned, with value
+        "dense" - series of values is returned with X.index and entry
+    annotation_values : str - one of "indicator" and "score"
+        "indicator" - value is bool = is this an outlier
+        "score" - value is float = outlier score
+    """
+
+    _tags = {
+        "handles-panel": False,  # can handle panel annotations, i.e., list X/Y?
+        "handles-missing-data": False,  # can handle missing data in X, Y
+        "annotation-type": "point",  # can be point, segment or both
+        "annotation-labels": "outlier",  # can be one of, or list-subset of
+        #   "label", "outlier", "change"
+    }
+
+    def __init__(
+        self, estimator, annotation_format="sparse", annotation_values="indicator"
+    ):
+
+        if annotation_format not in ["sparse", "dense"]:
+            raise ValueError('annotation_format must be "sparse" or "dense"')
```

Comment:
* Perhaps also worth shorting them to `format` and `values` since the annotation context is clear. 
* Should `values` be renamed to `labels`? Values may be a bit generic, labels will be familiar from the classification setting. 
* I suggest to write helper functions for input checks, e.g. `_check_format(format)`, and add them to `sktime/utils/validation/annotation.py`

Reply:
>since the annotation context is clear.

yes, but the context that this is something that applies to the output is not clear.
How about `y_format` etc?

Reply:
>I suggest to write helper functions for input checks,

With this design, would that not be identical with `check_series`?

Reply:
>Values may be a bit generic, labels will be familiar from the classification setting.

Yes, good idea

Reply:
it already did `check_series` when you wrote the above, so can you please explain what you would want on top of that?

Reply:
>should values be renamed to labels

there actually already is a variable that is called "labels", it's the one which tells you the type of labels

Reply:
For readability, I'd write:
```python
def check_fmt(fmt):
    valid_fmts = ["sparse", "dense"]
    if fmt not in valid_fmts:
        raise ValueError(f"`fm` must be in: {valid_fmts}, but found: {fmt}.")
    return fmt
```
We shouldn't use `format`, that's already a Python function.